count_cells_in_matrix.py

```diff
@@ -76,18 +84,41 @@ def get_project_paths() -> Sequence[str]:
     return project_paths
 
 
-def write_project_cell_count(project_path: str, cell_count: int):
-    """
-    Save the cell count to a file in the project path
-
-    :param project_path: A path to the project folder
-    :param cell_count: A count of cells
-    """
-    if cell_count is not None:
-        cell_count_file = f'{project_path}/cell_counts'
-        logging.info('Writing: %s', cell_count_file)
-        with open(cell_count_file, 'w') as f:
-            f.write(f'{cell_count}\n')
+def update_cell_count_file(accession_id: str, cell_count: int) -> bool:
+    if not accession_id or cell_count is None:
+        return False
+    cell_counts_file = 'cell_counts.json'
+    if os.path.exists(cell_counts_file):
+        with open(cell_counts_file, 'r') as f:
+            cell_counts = json.loads(f.read())
+    else:
+        cell_counts = {}
+    logging.info('Writing to cell counts json: accession=%s count=%s', accession_id, cell_count)
+    cell_counts[accession_id] = cell_count
+    with open(cell_counts_file, 'w') as f:
+        f.write(json.dumps(cell_counts, sort_keys=True, indent='    '))
+    return True
+
+
+def accession_id_from_project_path(project_path: str) -> str:
+    # Try to get the accession id from the project json file
+    # TODO: handle multiple accession ids in project json
+    project_json = f'{project_path}/bundle/project_0.json'
```

Comment:
This should be dependent on the GEO accession symlink and only that. I'd rather not depend on the presence of metadata (because that introduces an ordering dependency) or on the GEO file naming convention (because it's brittle).  Simply require that the argument is a symlink and get its name.

Reply:
I've reworked the download script to be accession id centric.